api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentService.java

```diff
@@ -142,17 +123,35 @@ public Payment makePayment(PaymentCycle paymentCycle, String cardAccountId) {
         } catch (RuntimeException e) {
             String failureMessage = String.format("Payment failed for cardAccountId %s, claim %s, paymentCycle %s, exception is: %s",
                     cardAccountId, paymentCycle.getClaim().getId(), paymentCycle.getId(), e.getMessage());
-            MakePaymentEvent failedEvent = buildFailedMakePaymentEvent(paymentCycle, payment, depositFundsResponse);
+            MakePaymentEvent failedEvent
+                    = buildFailedMakePaymentEvent(paymentCycle.getClaim(), paymentCycle.getTotalEntitlementAmountInPence(), payment, depositFundsResponse);
+            throw new EventFailedException(failedEvent, e, failureMessage);
+        }
+        return payment;
+    }
+
+    private Payment makePayment(PaymentCycle paymentCycle, String cardAccountId, int amountInPence) {
+        Payment payment = null;
+        DepositFundsResponse depositFundsResponse = null;
+        try {
+            payment = createPayment(paymentCycle, cardAccountId, amountInPence);
+            depositFundsResponse = depositFundsToCard(payment);
+            updatePayment(payment, depositFundsResponse.getReferenceId());
+            eventAuditor.auditMakePayment(paymentCycle, payment, depositFundsResponse);
+        } catch (RuntimeException e) {
+            String failureMessage = String.format("Payment failed for cardAccountId %s, claim %s, paymentCycle %s, exception is: %s",
+                    cardAccountId, paymentCycle.getClaim().getId(), paymentCycle.getId(), e.getMessage());
+            MakePaymentEvent failedEvent = buildFailedMakePaymentEvent(paymentCycle.getClaim(), amountInPence, payment, depositFundsResponse);
             throw new EventFailedException(failedEvent, e, failureMessage);
         }
         return payment;
     }
 
     @SuppressWarnings("PMD.NullAssignment")
-    private MakePaymentEvent buildFailedMakePaymentEvent(PaymentCycle paymentCycle, Payment payment, DepositFundsResponse depositFundsResponse) {
+    private MakePaymentEvent buildFailedMakePaymentEvent(Claim claim, int entitlementInPence, Payment payment, DepositFundsResponse depositFundsResponse) {
         return MakePaymentEvent.builder()
-                .claimId(paymentCycle.getClaim().getId())
-                .entitlementAmountInPence(paymentCycle.getTotalEntitlementAmountInPence())
+                .claimId(claim.getId())
+                .entitlementAmountInPence(entitlementInPence)
```

Comment:
This isn't quite right, should always be the entitlement amount, which it isn't except for the `makePaymentForPaymentCycle`.